numpy/_core/src/multiarray/convert_datatype.c

```diff
@@ -409,11 +409,12 @@ _get_cast_safety_from_castingimpl(PyArrayMethodObject *castingimpl,
  * implementations fully to have them available for doing the actual cast
  * later.
  *
- * @param from
+ * @param from The descriptor to cast from 
  * @param to The descriptor to cast to (may be NULL)
  * @param to_dtype If `to` is NULL, must pass the to_dtype (otherwise this
  *        is ignored).
- * @param[out] view_offset
+ * @param view_offset The byte offset, if any, between the original 
+ *        view/type and the view/type that would be cast to.
```

Comment:
```suggestion
 * @param view_offset If set, the cast can be described by a view with
 *        this byte offset.  For example, casting "i8" to "i8,"
 *        (the structured dtype) can be described with `*view_offset = 0`.
```
Maybe to make it a bit clearer, since this is a tricky concept.